src/dotty/tools/dotc/core/TypeParamInfo.scala

```diff
@@ -0,0 +1,29 @@
+package dotty.tools.dotc.core
+
+import Names.Name
+import Contexts.Context
+import Types.{Type, TypeBounds}
+
+/** A common super trait of Symbol and LambdaParam.
+ *  Used to capture the attributes of type parameters which can be implemented as either.
+ */
+trait TypeParamInfo {
+
+  /** Is this the info of a type parameter? Might be wrong for symbols */
```

Comment:
This should explain when and why this might be wrong for symbols, and if "wrong" means "true when it should be false" or "false when it should be true" or "both"